### DISCUSSION TITLE
[cli][build-storybook] Enable built assets to be output to a separate directory

### DISCUSSION THREAD

**Discussion by hydrosquall** [storybookjs/storybook#23087] [Ideas]
**Is your feature request related to a problem? Please describe**

- Our storybook is deployed and working with to a long URL. We'd like to deploy our built Storybook to a custom subdomain
- For this to happen, we need `index.html` to exist in a separate directory than the built `JS`  files. Currently, the built HTML and JS files are all in a flat directory structure. Our subdomain routing service needs all the built files to be in their own folder. Otherwise it can't differentiate between which paths should be rewritten to `index.html` for the client-side router to handle, and which paths should be served directly as astatic assets.

In other words, currently our build folder looks something like

```
index.html
iframe.html
main.hash.iframe.bundle.js
runtime~main.hash.js
static/
    image.png
```

We'd like our built folder to look something like this

```
index.html
iframe.html
assets/
    main.hash.iframe.bundle.js
    runtime~main.hash.js
static/
    image.png
```

**Describe the solution you'd like**

- Ideally, we'd have a command line flag in `build-storybook`  which only affects the built JS / CSS / any assets besides the base `index.html` and `iframe.html`, so that built assets can be deployed separately from the base HTML files.

```
build-storybook --assets assets/
```

**Describe alternatives you've considered**

- The parcel ecosystem addresses this issue through plugins: https://github.com/parcel-bundler/parcel/issues/233#issuecomment-599081474

**Are you able to assist to bring the feature to reality?**

- Yes, assuming this is something that makes sense to add / + with pointers about what files to modify

**Comment by shilman** [comment]
Converting this to a discussion. We’re reserving feature issues for things that are on our [short-term roadmap](https://github.com/storybookjs/storybook/milestones). We’ll check back periodically to see whether this is a good candidate to be added. If this feature would be useful to you, please upvote!
